File: swea/D2/2007.패턴 마디의 길이/2007.py
T = int(input())

for tc in range(1, T+1):
    s = input()

    for i in range(1, 11):
        l = 1
        flag = True

        for l in range(i, 30):
            if s[l] != s[l - i]:
                flag = False
                break
            l += 1
        
        if flag:
            print(f'#{tc} {i}')
            break

File: swea/D2/2007.패턴 마디의 길이/2007.cpp
#include <iostream>
#include <string>

using namespace std;

int T;

int main(){
  cin >> T;

  for(int t = 1; t <= T; t++){

    string s;
    cin >> s;
    
    for(int i = 1; i < 11; i++){
      bool flag = true;
      
      for(int l = i; l < 30; l++){
        if(s[l] != s[l - i]){
          flag = false;
          break;
        }
        
      }

      if(flag){
        cout << '#' << t << ' ' << i << endl;
        break;
      }
    }
  }
}